src/components/NewTodo/NewTodo.js

```diff
@@ -0,0 +1,209 @@
+  state = {
+    title: '',
+    name: '',
+    completed: '',
+    userId: '',
+    id: this.props.todoId,
+    status: '',
+    showTitleError: false,
+    showUserError: false,
+    showStatusError: false,
+  }
+
+  addTodoText = (event) => {
+    this.setState(
+      {
+        title: event.target.value.replace(/[^a-zа-я0-9\s]/g, ''),
+        showTitleError: false,
+      },
+    );
+  }
+
+  addUser = (id) => {
+    const { users } = this.props;
+
+    this.setState(
+      {
+        userId: +id,
+        name: users.find(user => user.id === +id).name,
+        showUserError: false,
+      },
+    );
+  }
+
+  statusTodo = (event) => {
+    const target = event.target.value;
+
+    this.setState({
+      completed: Boolean(+target),
+      status: target,
+      showStatusError: false,
+    });
+  }
+
+  reset = () => {
+    this.setState(prev => (
+      {
+        title: '',
+        name: '',
+        completed: '',
+        userId: '',
+        id: prev.id + 1,
+        status: '',
+        showTitleError: false,
+        showUserError: false,
+        showStatusError: false,
+      }
+    ));
+  }
+
+  sendTodo = (event) => {
+    const { name, userId, id, completed, status } = this.state;
+    let { title } = this.state;
+    const { addTodos } = this.props;
+
+    title = title.trim();
+
+    event.preventDefault();
+    if (title.length === 0) {
+      this.setState({
+        showTitleError: true,
+        showUserError: true,
+        showStatusError: true,
+      });
+
+      if (title) {
+        this.setState({
+          showTitleError: false,
+        });
+      }
+
+      if (userId) {
+        this.setState({
+          showUserError: false,
+        });
+      }
+
+      if (status) {
+        this.setState({
+          showStatusError: false,
+        });
+      }
+
+      return;
+    }
+
+    addTodos({
+      name,
+      title,
+      userId,
+      id,
+      completed,
+    });
+    this.reset();
+  }
+
+  render() {
+    const { users } = this.props;
+    const { title,
+      showTitleError,
+      userId,
+      showUserError,
+      status,
+      showStatusError } = this.state;
+
+    return (
+      <>
+        <form className="form" onSubmit={this.sendTodo}>
+          <label className="form__item">
+            <span className="form__name">Todo title</span>
+            <input
+              className={showTitleError
+                ? 'form__error'
+                : ''}
+              value={title}
+              onChange={this.addTodoText}
+              type="text"
+            />
+            {showTitleError
+              && (
+                <div className="form__error--message">
+                  Please enter the title
+                </div>
+              )
+            }
+          </label>
+          <br />
+          <label className="form__item">
+            <span className="form__name">Chose user</span>
+            <select
+              className={showUserError
```

Comment:
use classnames lib for this purpose, it's really convenient to use classed dynamically with it
https://www.npmjs.com/package/classnames